.buildkite/utils.sh

```diff
@@ -50,26 +50,36 @@ deploy_secrets_and_config() {
 
 run_migrations() {
   set -x
+
+  export IMAGE_SPEC="${DOCKER_REGISTRY}/${FLASK_APP}:${BUILDKITE_COMMIT}"
+
   aws eks --region us-east-2 update-kubeconfig --name ${EKS_CLUSTER}
   kubectl config set-context --current --namespace=${NAMESPACE}
-  cat migration-job.yml | sed "\
-        s/{{BUILDKITE_COMMIT}}/${BUILDKITE_COMMIT}/g;\
-        s/{{DOCKER_REGISTRY}}/${DOCKER_REGISTRY}/g;\
-        s/{{FLASK_APP}}/${FLASK_APP}/g" |
-    kubectl delete --ignore-not-found=true -f -
-  cat migration-job.yml | sed "\
-        s/{{BUILDKITE_COMMIT}}/${BUILDKITE_COMMIT}/g;\
-        s/{{DOCKER_REGISTRY}}/${DOCKER_REGISTRY}/g;\
-        s/{{FLASK_APP}}/${FLASK_APP}/g" |
-    kubectl apply -f -
+
+  sed "s|{{CONBENCH_WEBAPP_IMAGE_SPEC}}|${IMAGE_SPEC}|g" \
+    < k8s/conbench-database-migration.templ.yml \
+    > _jobspec
+
+  # Delete job first -- why is that important?
+  kubectl delete --ignore-not-found=true -f _jobspec
+  kubectl apply -f _jobspec
+
+  # Note(JP): we give this 24 hours of time. Why? For those heavy migration
+  # jobs that really take so long? Interesting.
```

Comment:
I forgot to add to the comment here that maybe this is not wise. Maybe we want this job to error out early, and maybe we should to one-off migrations that are known / expected to take forever manually.

Reply:
I do like the accountability & linking of the logs if we apply all migrations through this job, but I agree it's difficult to handle different sized loads. I wonder if there's any way that the person kicking off a deployment could configure a longer timeout than a default, shorter timeout.